### ISSUE TITLE
remove thread-unsafe static data

### ISSUE THREAD

**Issue by jeffhammond** [GlobalArrays/ga#3]
I have a very large number of patches related to thread-safety in https://github.com/jeffhammond/ga/tree/thread-safe.  This issue is a placeholder to track progress towards merging those.

**Comment by jeffdaily** [comment]
Thanks, Jeff.  If you had sent me your source tree beforehand I had lost it.  Thanks for putting it as an issue.

We recently discussed our development plan.  After GA-5.6 goes out with our MPI3 RMA port, we will be working on thread safety.  Hopefully your source tree will make short work of an initial implementation.  If I can't cherry-pick your source tree directly, I will otherwise attribute your changes as we make them.

**Comment by jeffdaily** [comment]
@jeffhammond I just clicked on your link and it gave me the github 404.

**Comment by jeffhammond** [comment]
Essentially all of my thread-safety work was in GA not ARMCI because I already implemented thread-safety in ARMCI-MPI (although this property is completely unverified).

**Comment by jeffhammond** [comment]
Sorry about the 404.  I added that link before I renamed that repo so that it wouldn't collide when I forked this repo.

**Comment by jeffdaily** [comment]
I found the new one.  No worries.
https://github.com/jeffhammond/ga-old/tree/thread-safe

Looks like we've got perhaps 44 commits to review.  Without having looked at them all yet, are any of them more important than others?

**Comment by jeffdaily** [comment]
If we have questions about a particular commit, should we comment on your commit in your ga-old tree or just inline here in this issue tracker?  Preference?

**Comment by jeffhammond** [comment]
Let me redo the changes.  I did a lot of fine-grain commits for my own purposes but it may be better to fuse them for review.

**Comment by jeffhammond** [comment]
In any case, I'll submit pull requests based upon my fork and we can review there.

**Comment by jeffdaily** [comment]
I don't yet have experience with github pull requests, sorry.  I'm trying to enforce a git-flow model, so is there a way to do the pull request off of our develop branch?  It's a bit of a mess right at the moment since we only just got onto github and are trying to polish a 5.6 release which will eventually get merged back into develop...

**Comment by jeffhammond** [comment]
Yeah, I can create a PR targeting any branch.  I haven't started redoing the changes yet so it is trivial to apply them to `develop` instead of `master`.

**Comment by jeffhammond** [comment]
I am finding it very hard to rebase because of all the build system crud that used to be in the repo, as well as some preprocessor symbol changes.  I will see if I can simplify things by squashing my commits.

**Comment by jeffdaily** [comment]
Let me know how I can help.  This will all be much easier soon.  Hopefully.  We're getting rid of the generated files from the repo.
